### ISSUE TITLE
Read smart values without needing collectd running as root

### ISSUE THREAD

**Issue by sqozz** [collectd/collectd#2143]
*   Version of collectd: 5.6.2.git
*   Operating system / distribution: Gentoo

## Expected behavior

The smart plugin should be able to read S.M.A.R.T.-values without the need for collect to run as the user "root". This is the default rollout of gentoo. I think, it's basically a good idea to run collectd as another user with caped capabilities. Please correct me, if i'm wrong and collectd is not supported to run as another user except root, then i'll report the bug at the gentoo-bugtracker. As stated as in the smartctl bugtracker (https://www.smartmontools.org/ticket/61), smartctl is not intended to run as other user than root. Is this the same case with the used libatasmart?

## Actual behavior

collectd running as user "collectd" (as provided by the service startup in gentoo) fails silently without any message while reading the  S.M.A.R.T.-values, even in the debug-log. Running collectd manually as user root, delivers data as excepted.
Also "collectdctl" does not provide any helpful information to debug this further for the user.

## Steps to reproduce

*   Enable the smart plugin in /etc/collectd.conf and ensure, at least one disk is included
*   Run collectd as other user then root
*   Watch "collectdctl" and see, how no datapoints are added for the smart values
*   Run collectd as user root
*   Take a look at collectdctl and see, how data points get added as expected

**Comment by mfournier** [comment]
Most plugins don't require root privileges to work. The few that do can usually be run as a standard user after settings capabilities properly (see `capabilities(7)` manpage).

```
$ collectd -C /tmp/collectd.conf -T
...
smart plugin: unable to open /dev/sda.
...
$ ls -l /dev/sda
brw-rw---- 1 root disk 8, 0 Jan 19 20:51 /dev/sda
$ sg disk
$ sudo setcap cap_sys_rawio=ep /usr/sbin/collectd
$ collectd -C /tmp/collectd.conf -T
...
write_log values:                                                                 
host.smart-sda.smart_badsectors 0 1485286242
```

NB: this plugin should log a warnings about this.

**Comment by sqozz** [comment]
This did indeed worked for me. Thanks for the useful information. I think this bug can be closed since you opened #2152 
Keep up the good work :+1:
